pandas/io/parsers.py

```diff
@@ -847,12 +847,18 @@ def _validate_usecols_arg(usecols):
     or strings (column by name). Raises a ValueError
     if that is not the case.
     """
+    msg = ("The elements of 'usecols' must "
+           "either be all strings or all integers")
+
     if usecols is not None:
         usecols_dtype = lib.infer_dtype(usecols)
-        if usecols_dtype not in ('integer', 'string'):
-            raise ValueError(("The elements of 'usecols' "
-                              "must either be all strings "
-                              "or all integers"))
+
+        if usecols_dtype == 'mixed':
```

Comment:
@jreback : What about a new `lib` method like `is_string_unicode_array` (similar to `is_integer_float_array`?  I don't see anything wrong with the `dtype` inference checks for `string` and `unicode` at the moment.

@Everyone: For anyone who doesn't quite know / remember where the source code for `infer_dtype` is, you can find it <a href="https://github.com/pydata/pandas/blob/master/pandas/src/inference.pyx#L83">here</a>.